### DISCUSSION TITLE
Triggering different channel from script in current channel

### DISCUSSION THREAD

**Discussion by areutera** [nextgenhealthcare/connect#6161] [Q&A] [answered]
Hello there!

For a project we are queuing certain HL7 messages (ADT / ORU / BAR) in separate channels (this is wip and not hewn into stone).

On recognizing a certain ADT-A03 message belongs to a case from a list of distinct cases, I would like to trigger a script in the ORU and BAR channels respectively to "start their thing" and execute scripts within their respective channels.

My colleagues running Mirth day-to-day tell me there is no easy way to do this, and looking through triggers / alert / events nothing immediate stood out to me. 

I can't imagine, though, there is no way to address another channel from within a script in Mirth.

And no, while I am reasonably proficient with JavaScript within such boundaries, I am a project manager, but not a coder per se.

Any ideas putting me on the right path would be very much appreciated!

**Comment by pacmano1** [comment]
What do you mean by start their thing?

Do you mean have the channel poll?

Do you mean that the channel is running all the time and when the condition is met you need some extra code to run ?

Either one is possible, but you need to describe your use case a bit more.

**Reply by areutera** [reply]
The second channel (current solution) is keeping the messages in the queue (at this point the amount seems manageable, rather than dumping everything in its own database). 

When the patient in question leaves the hospital we want to reprocess the queue for previous message content for this specific patient. 

Does this help to describe the "problem"?

The longer I explain this the more it seems probable to me, that dumping everything in a database seems to be the easier way... :(

**Reply by pacmano1** [reply]
In your use case I probably would dump it to a database.  It's likely the better way also.

**Reply by areutera** [reply]
Thank you for the quick reply! At least that helps me to steer things into the right direction.

I wasn't comfortable with keeping everything in the queue all that time anyways.

**Reply by pacmano1** [reply]
It is sort of an odd use case though - I might suspect your messages to always process realtime regardless of the patient dischage.

**Reply by jonbartels** [reply, marked as answer]
I think @pacmano1 has described the best solution - If messages are held, then use a separate database call to store those messages. The Mirth queue is meant for short-term message storage while other messages process, it is not meant for a longer delay waiting on external processes.

What the DB allows you to do is:

1. Recognize a message should be held and write it to the DB
2. When some other event happens, update the DB saying "stop holding this and send it"
3. Then your ORU and BAR channels can just poll the DB. `SELECT * FROM holding_tank WHERE ready_to_send = true` 
4. After sending your channels can then either UPDATE the status or DELETE the messages from the holding tank.

> My colleagues running Mirth day-to-day tell me there is no easy way to do this

Encourage your colleagues to read github, [the Mirth forums](https://forums.mirthproject.io/), and Mirth Slack. Even if something is hard to do in Mirth, there are people with 5-10-20 years of experience who are willing to help. Mirth has been around a very long time and others might have solved the problem already!

**Reply by areutera** [reply]
Once again, thanks for the quick replies. I'll pursue this angle now and chances are good that I'll be back with more questions, once I inevitably run into further roadblocks. ;-)
